torch/csrc/jit/script/compiler.cpp

```diff
@@ -1419,6 +1429,13 @@ struct to_ir {
     }
   }
 
+  Value* emitNone(SourceRange range) {
+    auto& g = *method.graph();
+    return g.insertNode(
+        g.create(prim::None, {}, 1)->setSourceLocation(
```

Comment:
You should add a `Operator` to `register_prim_ops.cpp` that implements `prim::None`. Like `prim::Undefined` but it should `return IValue()`.  This should never get emitted due to dead code elimination, but there may be some corner cases.